### DISCUSSION TITLE
Changes to Obsidian's installer

### DISCUSSION THREAD

**Discussion by rodrigoborgesdeoliveira** [microsoft/winget-pkgs#159993] [Q&A] [answered]
On [their latest update](https://github.com/obsidianmd/obsidian-releases/releases/tag/v1.6.5), Obsidian no longer has separate installers per scope and architecture. They now only have one installer. What is the right approach to update the package in winget? Some things to note:
- I'm not sure whether or not the same .exe works across x86, x64 and arm64, but I believe so. Is there an easy way to check this?
- Their installer now prompts whether you want to install for the current user or for all users - the current user is selected by default. Using the same installer url for both scopes results in both installing for the current user only (tested with `winget install ... --scope machine`).

**Comment by mdanish-kh** [comment]
> - I'm not sure whether or not the same .exe works across x86, x64 and arm64, but I believe so. Is there an easy way to check this?

You can specify x86 and x64 installers in the manifest and the pipelines will validate each one in their own respective environment. Unfortunately, if I recall correctly the pipelines don't have the ability to test arm installers on an arm architecture VM. That would have to be tested manually.

> - Their installer now prompts whether you want to install for the current user or for all users - the current user is selected by default. Using the same installer url for both scopes results in both installing for the current user only (tested with `winget install ... --scope machine`).

Is there a installer switch for specifying the scope? Some installers install in different scopes depending on whether the process is elevated (installed as admin) or non-elevated. Can you check if that's the case here?

**Reply by rodrigoborgesdeoliveira** [reply]
> Is there a installer switch for specifying the scope?

It seems like @SpecterShell was able to find them. :)

> Some installers install in different scopes depending on whether the process is elevated (installed as admin) or non-elevated. Can you check if that's the case here?

In an elevated shell, it was still getting installed under the user's scope. Thanks for pointing it out though. I'll keep that in mind for future contributions.

**Comment by SpecterShell** [comment, marked as answer]
https://github.com/microsoft/winget-pkgs/pull/160020
